libavoidrepetition.h

```diff
@@ -158,3 +158,43 @@ double min(double numbers[], int arrlen) {
 	}
 	return minitem;
 }
+
+
+char *input(char *str){
+    int bufsize = 100;
+    int position = 0;
+    char *buffer = malloc(sizeof(char) * bufsize);
+    int c;
+
+    if (!buffer) {
+        printf("\n\nERROR: ALLOCATION error\n\n");
+        exit(EXIT_FAILURE);
+    }
+
+    printf("%s", str);
+
+    while (1) {
+
+        c = getchar();
+
+        if (c == EOF || c == '\n') {
+            buffer[position] = '\0';
+            return buffer;
+        } else {
+            buffer[position] = c;
```

Comment:
Of course it isn't a bad practice to read the `stdin` byte-by-byte. The best practice, however, is to always use `fgets()`, optionally followed by `memset()`. 

I could merge it, but some more performance-loving people could complain. I think I'll commit something here. And hey, it's great code. You write shell code. So remember the comment I added on your shell post at reddit.